frontend/src/metabase/new_query/containers/MetricSearch.jsx

```diff
@@ -0,0 +1,97 @@
+import { fetchMetrics, fetchDatabases } from "metabase/redux/metadata";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import EntitySearch from "metabase/containers/EntitySearch";
+import { getMetadata, getMetadataFetched } from "metabase/selectors/metadata";
+import _ from 'underscore'
+
+import type { Metric } from "metabase/meta/types/Metric";
+import type Metadata from "metabase-lib/lib/metadata/Metadata";
+import EmptyState from "metabase/components/EmptyState";
+
+import type { StructuredQuery } from "metabase/meta/types/Query";
+import { getCurrentQuery } from "metabase/new_query/selectors";
+import { resetQuery } from '../new_query'
+
+const mapStateToProps = state => ({
+    query: getCurrentQuery(state),
+    metadata: getMetadata(state),
+    metadataFetched: getMetadataFetched(state)
+})
+const mapDispatchToProps = {
+    fetchMetrics,
+    fetchDatabases,
+    resetQuery
+}
+
+@connect(mapStateToProps, mapDispatchToProps)
+export default class MetricSearch extends Component {
+    props: {
+        getUrlForQuery: (StructuredQuery) => void,
+        query: StructuredQuery,
+        metadata: Metadata,
+        metadataFetched: any,
+        fetchMetrics: () => void,
+        fetchDatabases: () => void,
+        resetQuery: () => void,
+    }
+
+    componentDidMount() {
+        this.props.fetchDatabases() // load databases if not loaded yet
+        this.props.fetchMetrics(true) // metrics may change more often so always reload them
+        this.props.resetQuery();
+    }
+
+    getUrlForMetric = (metric: Metric) => {
+        const updatedQuery = this.props.query
+            .setDatabase(metric.table.db)
+            .setTable(metric.table)
+            .addAggregation(metric.aggregationClause())
+
+        return this.props.getUrlForQuery(updatedQuery);
+    }
+
+    render() {
+        const { metadataFetched, metadata } = this.props;
+        const isLoading = !metadataFetched.metrics || !metadataFetched.databases
+
+        return (
+            <LoadingAndErrorWrapper loading={isLoading}>
+                {() => {
+                    const sortedActiveMetrics = _.chain(metadata.metricsList())
+                        .filter((metric) => metric.isActive())
+                        .sortBy(({name}) => name.toLowerCase())
+                        .value()
+
+                    if (sortedActiveMetrics.length > 0) {
+                        return (
+                            <EntitySearch
+                                title="Which metric?"
+                                // TODO Atte Keinänen 8/22/17: If you call `/api/table/:id/table_metadata` it returns
+                                // all metrics (also retired ones) and is missing `is_active` prop. Currently this
+                                // filters them out but we should definitely update the endpoints in the upcoming metadata API refactoring.
+                                entities={sortedActiveMetrics}
```

Comment:
this should be a fairly quick backend endpoint to bang out, right @camsaul ?

though in general, I'd like to be a little saner about what gets pulled down onto the frontend redux store.